core/src/main/java/com/sap/ai/sdk/core/common/ClientResponseHandler.java

```diff
@@ -58,95 +60,83 @@ public ClientResponseHandler<T, E> objectMapper(@Nonnull final ObjectMapper jack
    *
    * @param response The response to process
    * @return A model class instantiated from the response
-   * @throws E in case of a problem or the connection was aborted
    */
   @Nonnull
   @Override
-  public T handleResponse(@Nonnull final ClassicHttpResponse response) throws E {
+  public T handleResponse(@Nonnull final ClassicHttpResponse response) {
     if (response.getCode() >= 300) {
-      buildExceptionAndThrow(response);
+      throw buildException(response);
```

Comment:
(Minor)

In Java the "expensive" part of exceptions is the implicit buildup of current stacktrace. That happens at exception instantiation, not when throwing. That also means this will look surprising that the stacktrace points at "new" and not at "throw" (here). Therefore, this line has a code smell.